crates/doc/src/inference.rs

```diff
@@ -1501,9 +1771,657 @@ fn regex_matches(re: &fancy_regex::Regex, text: &str) -> bool {
 mod test {
     use super::{super::Annotation, *};
     use json::schema::{self, index::IndexBuilder};
-    use serde_json::{json, Value};
+    #[cfg(test)]
+    use pretty_assertions::assert_eq;
+    use serde_json::{de, json, ser, Map, Value};
     use serde_yaml;
 
+    fn widening_snapshot_helper<T: AsRef<str>>(
+        initial_schema: Option<&str>,
+        expected_schema: &str,
+        docs: &[T],
+        enforce_limits: bool,
+    ) -> Shape {
+        let mut schema = match initial_schema {
```

Comment:
nit: it's a smell that we're repeating this all over the place. This should be a function.

Reply:
Yuuuup, thanks for pointing this out. Extracted into `Shape::invalid()`.